R/page_size.R

```diff
@@ -3,8 +3,63 @@
 NULL
 ## https://www.ietf.org/rfc/rfc0678.txt
 
-## This assumes fixed font size, monospaced font
+times_font_name <- function() {
+  plat <- .Platform$OS.type
+  switch(plat,
+    "unix" = "Times",
+    "windows" = "Times New Roman"
+  )
+}
 
+#' Font size specification
+#'
+#' @param font_family (`character(1)`)\cr font family to use during
+#'   string width and lines-per-page calculations. You can specify
+#'   "Times New Roman" as "Times" or "serif", regardless of OS.
+#'   Beyond that, see `family` entry in [graphics::par()]
+#'   for details.
+#' @param font_size (`numeric(1)`)\cr font size to use during string width
+#'   calculations and lines-per-page calculations.
+#' @param lineheight (`numeric(1)`)\cr line height to use during
+#'   lines-per-page calculations.
+#'
+#' @details Passing the output of this constructor
+#' to the rendering or pagination machinery defines
+#' a font for use when calculating word wrapping and pagination.
+#'
+#' @note Specifying font in this way to, e.g., [export_as_txt()] or
+#' [toString()] will not affect the font size of the output, as these
+#' are both raw text formats. [export_as_pdf()] will use the specified font.
+#'
+#' @seealso [nchar_ttype()], [toString()], [`pagination_algo`], [export_as_pdf()]
+#'
+#' @examples
+#' fspec <- font_spec("Courier", 8, 1)
+#'
+#' lets <- paste(letters, collapse = "")
+#'
+#' nchar_ttype(lets, fspec)
+#'
+#' fspec2 <- font_spec("Times", 8, 1)
+#'
+#' nchar_ttype(lets, fspec2)
+#'
+#' @export
+font_spec <- function(font_family = "Courier",
+                      font_size = 8,
+                      lineheight = 1) {
+  if (font_family %in% c("Times New Roman", "Times", "serif")) {
+    font_family <- times_font_name()
+  }
+  structure(
+    list(
+      family = font_family,
+      size = font_size,
+      lineheight = lineheight
+    ),
+    class = c("font_spec", "list")
+  )
+}
 std_cpi <- 10L
 std_lpi <- 6L
```

Comment:
There were already some default values. Should we group these somewhere more visible? even only the top of the script with some description of their uses

Reply:
These are ancient and unused (and also, you'll note, unchanged in the diff)They come from a *very* early initial pass at supporting page sizes. They can be removed, but this PR is probably technically not the right place to do so.

Reply:
it makes sense. I will create a new issue